packages/pages/src/vite-plugin/build/closeBundle/closeBundle.ts

```diff
@@ -44,8 +46,29 @@ export default (projectStructure: ProjectStructure) => {
 
     finisher = logger.timedLog({ startLog: "Validating functions" });
     try {
-      // loading the modules checks for all required aspects
-      await loadFunctions(projectStructure.functionBundleOutputRoot.path);
+      const functionFilepaths = getFunctionFilepaths("dist/functions");
+      await Promise.all(
+        functionFilepaths.map(async (filepath) => {
+          const buildResult = await esbuild.build({
```

Comment:
You shouldn't need to run esbuild. The files have already been built at this point. We should just import them directly like we do in loader.ts to validate the template modules. You'll see that we we call it will transpile=false in `templateModules = await loadTemplateModules(serverBundles, false, true);`